src/viur/core/skeleton.py

```diff
@@ -1968,34 +1968,60 @@ def canCall(self) -> bool:
         return user is not None and "root" in user["access"]
 
     def dataSkel(self):
-        modules = ["*"] + listKnownSkeletons()
-        modules.sort()
         skel = BaseSkeleton().clone()
-        skel.module = SelectBone(descr="Module", values={x: translate(x) for x in modules}, required=True)
+
+        skel.skelname = SelectBone(
+            descr="Kind",
+            values={"*": "*"} | {name: translate(name) for name in sorted(listKnownSkeletons())},
+        )
+
+        skel.specific_key = StringBone(
+            descr="Key",
+            params={
+                "tooltip": "Optional: Key of one entry to refresh, ignores the kind selected above."
+            }
+        )
+
         return skel
 
-    def execute(self, module, *args, **kwargs):
+    def execute(self, skelname, specific_key, *args, **kwargs):
         usr = current.user.get()
         if not usr:
             logging.warning("Don't know who to inform after rebuilding finished")
             notify = None
         else:
             notify = usr["name"]
 
-        if module == "*":
-            for module in listKnownSkeletons():
-                logging.info("Rebuilding search index for module %r", module)
-                self._run(module, notify)
+        if skelname == "*":
+            if specific_key:
+                raise errors.BadRequest(f"Cannot run on all kinds with {specific_key=}")
+
+            for skelname in listKnownSkeletons():
+                logging.info(f"Rebuilding search index for {skelname=}")
+                self._run(skelname, notify)
         else:
-            self._run(module, notify)
+            if specific_key:
+                try:
+                    specific_key = db.Key.from_legacy_urlsafe(specific_key)
+                    skelname = specific_key.kind
+                except Exception:
+                    pass  # just ignore, and use value of specific_key further.
+
+            if not skelname:
+                raise errors.BadRequest("Cannot run on unknown kind")
+
+            self._run(skelname, notify, **{"key": str(specific_key)})
```

Comment:
I think it's superfluous to allocate a dict and then unpack it directly, even though it can also be used as a direct argument.
```suggestion
            self._run(skelname, notify, key=str(specific_key))
```